pkg/cli/cmd/radinit/recipe.go

```diff
@@ -28,14 +27,90 @@ import (
 	msg_ctrl "github.com/radius-project/radius/pkg/messagingrp/frontend/controller"
 	recipe_types "github.com/radius-project/radius/pkg/recipes"
 	"github.com/radius-project/radius/pkg/to"
+	"github.com/radius-project/radius/pkg/ucp/ucplog"
 	"github.com/radius-project/radius/pkg/version"
 	"oras.land/oras-go/v2/registry/remote"
 )
 
 const (
-	DevRecipesRegistry = "ghcr.io/radius-project"
+	// RecipeRepositoryPrefix is the prefix for the repository path.
+	RecipeRepositoryPrefix = "ghcr.io/radius-project/recipes/local-dev/"
 )
 
+type DevRecipe struct {
+	// NormalizedName is the normalized name of the recipe.
+	//
+	// For example, "mongodatabases".
+	NormalizedName string
+
+	// ResourceType is the resource type of the recipe.
+	//
+	// For example, "Applications.Datastores/mongoDatabases".
+	ResourceType string
+
+	// RepoPath is the repository path of the recipe.
+	//
+	// For example, "ghcr.io/radius-project/recipes/local-dev/mongodatabases".
+	RepoPath string
+}
+
+// NewDevRecipe creates a new DevRecipe object and returns it.
+func NewDevRecipe(normalizedName, resourceType, repoPath string) *DevRecipe {
+	return &DevRecipe{
+		NormalizedName: normalizedName,
+		ResourceType:   resourceType,
+		RepoPath:       repoPath,
+	}
+}
```

Comment:
Do you think this is needed? In Go, types don't need a "constructor" like to in a language like C# or Java. It's simpler to leave this out and create the struct directly.